File: src/main/java/ru/hasa/springbootapp/service/ApiRestUserService.java
package ru.hasa.springbootapp.service;

import ru.hasa.springbootapp.dto.RoleDto;
import ru.hasa.springbootapp.dto.UserDto;

import java.util.List;

public interface ApiRestUserService {

    List<UserDto> getAllUsers();

    UserDto getUserById(Long id);

    UserDto saveUser(UserDto userDto);

    UserDto updateUser(UserDto userDto);

    void deleteUser(Long id);

    List<RoleDto> getAllRoles();
}

File: src/main/kotlin/ru/hasa/springbootapp/service/ApiRestUserService.kt
package ru.hasa.springbootapp.service

import ru.hasa.springbootapp.dto.RoleDto
import ru.hasa.springbootapp.dto.UserDto

interface ApiRestUserService {
    fun getAllUsers(): List<UserDto>
    fun getUserById(id: Long): UserDto
    fun saveUser(userDto: UserDto): UserDto
    fun updateUser(userDto: UserDto): UserDto
    fun deleteUser(id: Long)
    fun getAllRoles(): List<RoleDto>
}
